src/microbe_stage/Patch.cs

```diff
@@ -180,23 +185,21 @@ public class PatchSnapshot : ICloneable
     public double TimePeriod;
 
     public Dictionary<Species, long> SpeciesInPatch = new Dictionary<Species, long>();
+    public Dictionary<Species, SpeciesInfo> RecordedSpeciesInfo = new Dictionary<Species, SpeciesInfo>();
 
     public BiomeConditions Biome;
 
     public object Clone()
     {
+        // We only do a shallow copy of RecordedSpeciesInfo here, but SpeciesInfo are never modified.
         var result = new PatchSnapshot
         {
             TimePeriod = TimePeriod,
-            SpeciesInPatch = new Dictionary<Species, long>(SpeciesInPatch.Count),
+            SpeciesInPatch = new Dictionary<Species, long>(SpeciesInPatch),
+            RecordedSpeciesInfo = new Dictionary<Species, SpeciesInfo>(RecordedSpeciesInfo),
             Biome = (BiomeConditions)Biome.Clone(),
         };
 
-        foreach (var entry in SpeciesInPatch)
-        {
-            result.SpeciesInPatch.Add(entry.Key, entry.Value);
-        }
```

Comment:
This was changed to `SpeciesInPatch = new Dictionary<Species, long>(SpeciesInPatch),` I assume that properly copies the info in the dictionary? For some reason in the past I've written code like this in multiple places in Thrive, I don't remember why I didn't use that shorter code...

Reply:
From what I understand it does. I was surprised as well by the code, perhaps it simply went under the radar? I don't see any reason not to use the other method.

Reply:
If it works then it is good (and copies the data the same way as this explicit .Add call). And I don't see a problem with converting the remaining code also to use this approach, though that's probably best saved to a separate small cleanup PR.